packages/freezed/lib/src/templates/prototypes.dart

```diff
@@ -12,6 +12,32 @@ List<String> parseDecorators(List<ElementAnnotation> metadata) {
   ];
 }
 
+String variable(String name) {
+  return name.contains(r'$') ? '\${$name}' : '\$$name';
+}
+
+String encodeVariable(String name) {
+  final buffer = StringBuffer();
```

Comment:
Is there a reason why we can't do `return name.replaceAll(r'$' ,r'\$')`?

Reply:
Of course there is! If the input string is "a\$$$\$b\$" you? Or accidentally call the wrapper repeatedly! I changed the function name (accurately expressing the intent of the function) in the latest code to be compatible with multiple calls.

Reply:
Why would the input contain `\$` in it? I don't think that's a valid character for a property name, right?